### ISSUE TITLE
Underscores stripped from heading slugs

### ISSUE THREAD

**Issue by treyhunner** [rvben/rumdl#481]
### rumdl version

0.1.36

### Installation method

Pre-built binary

### Bug description

Underscores appear to be stripped when generating heading slugs, so a header containing the (backtick-escaped) `__hello__` would be seen as producing a slug like `hello` instead of `__hello__`, causing MD051 to fire on `#__hello__` links.

### Steps to reproduce

Here is `example.md`:
```markdown
## Introduction

### `__hello__`

The `__hello__` module prints "Hello world!"

## Summary

- This should match: [`__hello__`](#__hello__)
- This should be a problem: [`hello`](#hello)
```
Run this:
```bash
$ rumdl check --disable MD013,MD041 example.md
example.md:9:22: [MD051] Link anchor '#__hello__' does not exist in document headings

Issues: Found 1 issues in 1 file (11ms)
```

### Expected behavior

The link `#__hello__` should match the heading `__hello__`.

The link `#hello` should be flagged as non-existent.

### Actual behavior

The link `#hello` matches the heading `__hello__`.

The link `#__hello__` does not match the heading `__hello__`.

### Configuration

```toml

```

### Operating system

Linux

**Comment by rvben** [comment]
Fixed in [v0.1.37](https://github.com/rvben/rumdl/releases/tag/v0.1.37). Underscores inside code spans are now preserved during GitHub anchor slug generation. Thanks for reporting @treyhunner!
